src/grad.jl

```diff
@@ -85,4 +85,14 @@ end
 
 Compute the gradient of `f` for any `xs` for which [`to_vec`](@ref) is defined.
 """
-grad(fdm, f, xs...) = j′vp(fdm, f, 1, xs...)  # `j′vp` with seed of 1
+grad(fdm, f, xs...) = j′vp(fdm, f, 1.0, xs...)  # `j′vp` with seed of 1
+
+# This deals with the fact that integers are non perturbable
+# v, b = to_vec(1);
+# v == []
+# b(v) == 1
+# which means that jvp always returns the integer itself, since [] - [] == []
+_int2zero(x) = x
```

Comment:
I suspect that this will miss a few cases (e.g. vectors of integers), but probably that's fine for now -- we can revisit later it needs be.

Reply:
On a more general note: is the need for this function a manifestation of the fact that FiniteDifferences doesn't really know how to handle tangents properly?